src/core/Buttons/Button/Button.test.tsx

```diff
@@ -10,7 +10,13 @@ import { Button } from './Button';
 it('renders default button correctly', () => {
   const onClickMock = jest.fn();
   const { container } = render(
-    <Button onClick={onClickMock}>Click me!</Button>,
+    <Button
+      onClick={(e) => {
```

Comment:
@veekeys I created a branch for you to show how Ben's approach doesn't have correct types for events. All the code is in [`Button.tsx`](https://github.com/iTwin/iTwinUI-react/blob/mayank/polymorphic-types-ben/src/core/Buttons/Button/Button.tsx#L40). Check out that branch and open `Button.test.tsx:14` to see the `any` type.